lbrynet/file_manager/EncryptedFileManager.py

```diff
@@ -86,53 +85,49 @@ def _add_to_sd_identifier(self):
-        yield self._save_lbry_file(stream_hash, rate)
-
-    @defer.inlineCallbacks
-    def _check_stream_info_manager(self):
-        def _iter_streams(stream_hashes):
-            for stream_hash in stream_hashes:
-                yield self._check_stream_is_managed(stream_hash)
-
-        stream_hashes = yield self.stream_info_manager.get_all_streams()
-        log.debug("Checking %s streams", len(stream_hashes))
-        yield defer.DeferredList(list(_iter_streams(stream_hashes)))
-
     @defer.inlineCallbacks
     def _start_lbry_files(self):
-        yield self._check_stream_info_manager()
         files_and_options = yield self._get_all_lbry_files()
-        yield defer.DeferredList([
-            self._set_options_and_restore(rowid, stream_hash, options)
-            for rowid, stream_hash, options in files_and_options
-        ])
-
+        stream_infos = yield self.stream_info_manager._get_all_stream_infos()
+        b_prm = self.session.base_payment_rate_manager
+        payment_rate_manager = NegotiatedPaymentRateManager(b_prm, self.session.blob_tracker)
+        log.info("Trying to start %i files", len(stream_infos))
+        for i, (rowid, stream_hash, blob_data_rate, status) in enumerate(files_and_options):
+            if len(files_and_options) > 500 and i % 500 == 0:
+                log.info("Started %i/%i files", i, len(stream_infos))
+            if stream_hash in stream_infos:
+                if stream_infos[stream_hash]['suggested_file_name']:
+                    file_name = os.path.basename(stream_infos[stream_hash]['suggested_file_name'])
+                else:
+                    file_name = os.path.basename(stream_infos[stream_hash]['stream_name'])
+
+                lbry_file = ManagedEncryptedFileDownloader(
+                    rowid,
+                    stream_hash,
+                    self.session.peer_finder,
+                    self.session.rate_limiter,
+                    self.session.blob_manager,
+                    self.stream_info_manager,
+                    self,
+                    payment_rate_manager,
+                    self.session.wallet,
+                    self.download_directory,
+                    file_name=file_name,
+                    sd_hash=stream_infos[stream_hash]['sd_hash'],
+                    key=stream_infos[stream_hash]['key'],
+                    stream_name=stream_infos[stream_hash]['stream_name'],
+                    suggested_file_name=stream_infos[stream_hash]['suggested_file_name']
+                )
+                try:
+                    # restore will raise an Exception if status is unknown
+                    lbry_file.restore(status)
+                except Exception:
+                    log.warning("Failed to start %i", rowid)
+                    continue
+                self.lbry_files.append(lbry_file)
+        log.info("Started %i lbry files", len(self.lbry_files))
         if self.auto_re_reflect is True:
```

Comment:
I think  that start_lbry_file() function immediately below this should create ManagedEncryptedFileDownloader() in the same manner as _start_lbry_files(). Otherwise we have two completely different ways of creating ManagedEncryptedFileDownloader instances. 

This should mean that we should be able to delete some functions in the base classes of ManagedEncryptedFileDownloader like https://github.com/lbryio/lbry/blob/8eba9dc4b1b2a945fc66b7c5aab7354aad05ad50/lbrynet/lbry_file/client/EncryptedFileDownloader.py#L32 

that on its own pulls things from the stream_info_manager. 

Overall I think its good that the file manager is the one and only place responsible for pulling in data from stream_info_manager.